frontend/catalyst/debug/compiler_functions.py

```diff
@@ -160,3 +161,43 @@ def compile_from_mlir(ir, compiler=None, compile_options=None):
         result_types = [mlir.ir.RankedTensorType.parse(rt) for rt in func_data[1].split(",")]
 
     return CompiledFunction(shared_object, qfunc_name, result_types, None, compiler.options)
+
+
+@debug_logger
+def compile_cmain(fn, *args):
+    """Generate and compile a C program that calls a jitted function with the provided arguments.
+
+    Args:
+        fn (QJIT): a qjit-decorated function
+        *args: argument values to use in the C program when invoking ``fn``
+
+    Returns:
+        (str): the paths that should be included in LD_LIBRARY_PATH.
+        (str): the path of output binary.
+    """
+    f_name = str(fn.__name__)
+    workspace = str(fn.workspace)
+    main_c_file = workspace + "/main.c"
+    output_file = workspace + "/" + f_name + ".out"
+    shared_object_file = workspace + "/" + f_name + ".so"
+    lib_shared_object_file = workspace + "/lib" + f_name + ".so"
+    options = fn.compiler.options
+    with open(main_c_file, "w", encoding="utf-8") as file:
+        file.write(get_cmain(fn, *args))
+    shutil.copy(shared_object_file, lib_shared_object_file)
+
+    # configure flags
+    default_flags = LinkerDriver.get_default_flags(options)
+    no_shared_flags = [fs for fs in default_flags if fs != "-shared"]
+    link_so_flags = no_shared_flags + [
+        "-Wl,-rpath," + workspace,
+        "-L" + workspace,
+        "-l" + f_name,
+        "-g",
+    ]
+    LinkerDriver.run(main_c_file, outfile=output_file, flags=link_so_flags, options=options)
+
+    # generate ld library paths
+    lib_strings = [s[2:] for s in link_so_flags if s.startswith("-L")]
+    ld_env = "$LD_LIBRARY_PATH:" + ":".join(lib_strings)
+    return ld_env, output_file
```

Comment:
This is a bit concerning, do we have to set environment variables for the executable to work? For one, `LD_LIBRARY_PATH` won't work on mac, but normally that's what the RPATH is for anyway.

Reply:
If we do not run the executable with  `LD_LIBRARY_PATH`, we will get this error. I believe this is also related to conda environment. And `rpath`  somehow does not work with that. Will try to find any way get rid of `LD_LIBRARY_PATH`.
```c++
error while loading shared libraries: libgfortran-040039e1.so.5.0.0: cannot open shared object file: No such file or directory\
```

Reply:
I found I have to import gfortran and quadmath from scipy (like openblas).